manifests/install.pp

```diff
@@ -37,21 +36,73 @@
       'AIX'   => concat(['--ignoreos'],$install_options),
       default => $install_options
     }
-    $_package_options = {
+
+    # package provider does not provide 'versionable'
+    package { $::puppet_agent::package_name:
+      ensure          => 'present',
       provider        => 'rpm',
       source          => "/opt/puppetlabs/packages/${package_file_name}",
       install_options => $_install_options,
     }
   } elsif $::operatingsystem == 'Solaris' and $::operatingsystemmajrelease == '10' {
```

Comment:
There is a possibility the upgrade is succeeding but the Puppet process is killed immediately after. I'll try to look into that.